dtonator/src/main/java/com/bizo/dtonator/GenerateEnum.java

```diff
@@ -44,6 +49,27 @@ private void addEnumValues() {
     }
   }
 
+  private void addEnumGetters() throws ClassNotFoundException {
+    final Class<?> clazz = Class.forName(dto.getDomainType());
+    for (final Method method : clazz.getDeclaredMethods()) {
+      if (method.getName().startsWith("get")) {
+        final GMethod getter = gc.getMethod("{}", method.getName()).returnType(method.getReturnType());
+        getter.body.line("switch (this) {");
+        for (final String value : dto.getEnumValues()) {
+          @SuppressWarnings({ "unchecked", "rawtypes" })
+          final Enum enumInstance = Enum.valueOf((Class<? extends Enum>) clazz, value);
+          try {
+            final Object returnValue = method.invoke(enumInstance);
+            getter.body.line("_ case {}: return {};", value, returnValue instanceof Integer ? returnValue : "\"" + returnValue.toString() + "\"");
+          } catch (IllegalAccessException | InvocationTargetException ignore) {
+          }
+        }
+        getter.body.line("}");
+        getter.body.line("return null;");
```

Comment:
Well, this is annoying, but technically if we had a method, e.g. getFoo() that was going to blow up, we've already output half a method into the GClass; so this should probably try to call all of the values first, and then assuming no errors happen for all of the enum values, go ahead and output the whole method.